ember-cli/2017-03/march-31.md

```diff
@@ -0,0 +1,127 @@
+# ember-cli Meeting Notes - 2017-03-31
+
+## Attendees
+
+- Nathan Hammond
+- Jake Bixby
+- Tobias Bieniek
+- David Hamilton
+- Alex Navasardyan
+- Katie Gengler
+- Kelly Selden
+- Chad Hietala
+- Robert Jackson
+
+## Agenda
+
+- Glimmer Pipeline
+- Tree Shaking
+- `app.import()` from `node_modules`
+- Blueprint (`blprnt`)
+  - use Glimmer resolver?
+  - composability of blueprints
+  - blueprint migrations/transactional?
+- Background Service
+- Slow Commands
+  - `generate`
+  - `help`
+- Ember CLI/Broccoli Errors
+- Global add-ons
+- `lib-cli`
+
+## Discussion Notes
+
+### Glimmer Pipeline
+
+Extract `lib/broccoli/ember-app.js` into a separate npm package.
+
+### Tree Shaking
+
+- given list of file entries + module information + roots, generate a retained graph
+- keep module loose w/o concat until the very end
+- generate dep graph at the very end of the build process
+- have a hook that will give you access to dep graph w/o being able to modify it
+
+### `app.import` from `node_modules`
+
+- support amd, cjs or es6 w/o watching
+- everything that's imported is considered dynamically linked
+- should care about single file (entry point) make sure we can tell the difference between globals/modules for tree shaking (globals and module are in different buckets)
+- need to be able to "blacklist" modules to resolve them during runtime
+  - possibly make it automatic, we *should* have info already or do a dep walk on app.import path
+  - should *not* require for users to pass into list of modules when using `app.import`
+- address differences between `import`s (documentation or otherwise)
+  - `app.import` is project relative
+  - `this.import` is relative to whatever `this` is
+- path *has* to start with `vendor`, `node_modules`
+
+### Blrpnt extraction
+
+- unfork
+  - port new test changes from ember-cli
+  - make sure we haven't lost anything added in ember-cli/blueprint.js
+- load blprnt in ember-cli
+  - bring into ember-cli as drop in replacement
+- decouple pods/classic structures information from blprint/blueprint.js
+  - create new structures using new primitive to manage this logic in ember-cli
+- create module unification structure to provide blueprint support
+  - use glimmer resolver to take advantage of rules to generate file paths
+
+### Background Service
+
+**idea 1**: Convert bin/cli into an actor system to provide parallelism  (actor === task)
+
+**idea 2**: Daemonize the process
+
+- removing add-ons (packages.json changes) needs to restart daemon
+- daemon per application
+- abstraction at app layer
+
+*Conclusion*: Since the gain of this refactor doesn't seem high enough
```

Comment:
i think we should just make commands load quickly. I don't believe this is unreasonable. The issue today is just that command files typically eager load stuff. 

We should give them a budget and warn if they exceed /w docs on how to do this.